File: branches/Medusa/Source/Fantasy.Jobs/Resources/JobRuntimeScheduleResourceProvider.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Fantasy.Jobs.Scheduling;
using Fantasy.Jobs.Management;

namespace Fantasy.Jobs.Resources
{
    public class JobRuntimeScheduleResourceProvider : RuntimeScheduleResourceProviderBase
    {
       

        protected override bool InternalCanHandle(string name)
        {
            return string.Equals("RunJob", name, StringComparison.OrdinalIgnoreCase);
        }

        public override void Initialize()
        {
            base.Initialize();
            JobManagerSettings.Default.PropertyChanged += new System.ComponentModel.PropertyChangedEventHandler(JobManagerSettings_PropertyChanged);
        }

        void JobManagerSettings_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
        {
            if(e.PropertyName == "JobRuntimeSchedule")
            {
                this.Reschedule();
                
            }
        }

        protected override RuntimeScheduleSetting GetSetting(string id)
        {
            return JobManagerSettings.Default.JobRuntimeSchedule; 
        }

        protected override string GetResourceId(ResourceParameter parameter)
        {
            return "";
        }
    }
}

File: branches/Stheno/AutoConverted/Fantasy.Jobs/Resources/JobRuntimeScheduleResourceProvider.java
package Fantasy.Jobs.Resources;

import Fantasy.Jobs.Scheduling.*;
import Fantasy.Jobs.Management.*;

public class JobRuntimeScheduleResourceProvider extends RuntimeScheduleResourceProviderBase
{


	@Override
	protected boolean InternalCanHandle(String name)
	{
		return String.equals("RunJob", name, StringComparison.OrdinalIgnoreCase);
	}

	@Override
	public void Initialize()
	{
		super.Initialize();
//C# TO JAVA CONVERTER TODO TASK: Java has no equivalent to C#-style event wireups:
		JobManagerSettings.Default.PropertyChanged += new System.ComponentModel.PropertyChangedEventHandler(JobManagerSettings_PropertyChanged);
	}

	private void JobManagerSettings_PropertyChanged(Object sender, System.ComponentModel.PropertyChangedEventArgs e)
	{
		if(e.PropertyName.equals("JobRuntimeSchedule"))
		{
			this.Reschedule();

		}
	}

	@Override
	protected RuntimeScheduleSetting GetSetting(String id)
	{
		return JobManagerSettings.getDefault().getJobRuntimeSchedule();
	}

	@Override
	protected String GetResourceId(ResourceParameter parameter)
	{
		return "";
	}
}